error-handling/src/main/java/com/palantir/remoting1/errors/SerializableError.java

```diff
@@ -37,36 +36,30 @@
 
     public abstract String getMessage();
 
+    // TODO(rfink) This needs to get renamed once a wire-break is acceptable.
     @JsonProperty("exceptionClass")
-    public abstract String getExceptionClassName();
-
-    /**
-     * @deprecated Use {@link #getExceptionClassName()} instead.
-     */
-    @Deprecated
-    @JsonIgnore
-    @SuppressWarnings("unchecked")
-    public final Class<? extends Exception> getExceptionClass() {
-        try {
-            return (Class<? extends Exception>) Class.forName(getExceptionClassName());
-        } catch (ClassNotFoundException e) {
-            throw new RuntimeException(e);
-        }
-    }
+    public abstract String getExceptionName();
 
     @Nullable
     public abstract List<StackTraceElement> getStackTrace();
 
+    /** Constructs a new error whose exception name is the fully-qualified name of the given class. */
     public static SerializableError of(String message, Class<? extends Exception> exceptionClass) {
```

Comment:
I'll do the latter.

On Tue, Aug 16, 2016 at 6:24 AM Mark Elliot <notifications@github.com>
wrote:

> In
> error-handling/src/main/java/com/palantir/remoting1/errors/SerializableError.java
> <https://github.com/palantir/http-remoting/pull/199#discussion_r74932836>:
>
> >
> >      @Nullable
> >      public abstract List<StackTraceElement> getStackTrace();
> >
> > +    /** Constructs a new error whose exception name is the fully-qualified name of the given class. */
> >      public static SerializableError of(String message, Class<? extends Exception> exceptionClass) {
>
> I'm basically flagging that the TODO above which changes exceptionClass
> to something like errorName in some eventual wire-format break should see
> a corresponding change here. We should probably start encouraging use of
> error names instead of exception classes wherever possible. OTOH, I could
> see strictly adding one more method here and leaving this one (maybe
> deprecated?) for convenience.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/palantir/http-remoting/pull/199/files/d620e6e9534492bdaf02de0938714cb37e370b24#r74932836>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGOdweqHmC7fqdQYG1Bo9IXOrmLBHd_5ks5qgbnmgaJpZM4Jjywx>
> .
>